### PR TITLE
fix(streaming): handle nil taker order entities

### PR DESCRIPTION
**Pull request by UnbornAztecKing**
### Changelist
Check for nil taker order entities at each level when filtering taker orders for streaming.

### Test Plan
Added unit tests for check routine.
Deployment to internal networks and testnet.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved null-safety in streaming update filtering to prevent edge-case failures when order or taker data is missing; call sites updated to use the safer filter.

* **Tests**
  * Added comprehensive tests for order filtering (covers nil, missing, matching and non-matching cases).
  * Note: a duplicate test definition was introduced and should be cleaned up.

### CODE REVIEW

File: protocol/streaming/full_node_streaming_manager_test.go [hunk 1]
```diff
@@ -605,3 +606,88 @@ func TestFilterStreamUpdatesWithDuplicateSubaccountIds(t *testing.T) {
...
+			shouldFilter: false,
+		},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			shouldFilter := streaming.DoFilterTakerOrderBySubaccount(tc.takerOrder, tc.subaccountIds)
+			require.Equal(t, tc.shouldFilter, shouldFilter)
+		})
+	}
+}
```
[the remark this answers is not in the document]
Comment (UnbornAztecKing) [reply protocol/streaming/full_node_streaming_manager_test.go:709]: check again
